modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -103,64 +108,68 @@ public void testSendNotification() throws Exception {
-			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
-			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
-			RandomTestUtil.randomString(),
-			ObjectActionExecutorConstants.KEY_NOTIFICATION,
-			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
-			UnicodePropertiesBuilder.put(
-				"notificationTemplateId",
-				notificationTemplate.getNotificationTemplateId()
-			).build());
-
-		_addObjectEntry();
-
-		notificationQueueEntries = ListUtil.sort(
-			notificationQueueEntryLocalService.getNotificationEntries(
-				NotificationConstants.TYPE_EMAIL,
-				NotificationQueueEntryConstants.STATUS_SENT),
-			Comparator.comparing(
-				notificationQueueEntry -> {
-					Map<String, Object> notificationRecipientSettingsMap =
-						NotificationRecipientSettingUtil.
-							getNotificationRecipientSettingsMap(
-								notificationQueueEntry);
+		objectActionLocalService.deleteObjectAction(objectAction);
 
-					return String.valueOf(
-						notificationRecipientSettingsMap.get("to"));
-				}));
-
-		Assert.assertEquals(
-			notificationQueueEntries.toString(), 2,
-			notificationQueueEntries.size());
-
-		List<String> expectedToEmailAddresses = ListUtil.sort(
+		List<String> expectedToEmailAddresses1 = ListUtil.sort(
 			Arrays.asList(user1.getEmailAddress(), user2.getEmailAddress()));
 
-		_assertNotificationQueueEntry(
-			true, expectedToEmailAddresses.get(0),
-			notificationQueueEntries.get(0));
-		_assertNotificationQueueEntry(
-			true, expectedToEmailAddresses.get(1),
-			notificationQueueEntries.get(1));
+		_testSendNotificationWithToProperty(
+			expectedToEmailAddresses1,
+			StringBundler.concat(
+				user1.getEmailAddress(), StringPool.COMMA,
+				user2.getEmailAddress()));
 
-		for (NotificationQueueEntry notificationQueueEntry :
-				notificationQueueEntries) {
+		_testSendNotificationWithToProperty(
+			expectedToEmailAddresses1,
+			StringBundler.concat(
+				user1.getEmailAddress(), StringPool.COMMA_AND_SPACE,
+				user2.getEmailAddress()));
 
-			notificationQueueEntryLocalService.deleteNotificationQueueEntry(
-				notificationQueueEntry);
-		}
+		_testSendNotificationWithToProperty(
+			expectedToEmailAddresses1,
+			StringBundler.concat(
+				user1.getEmailAddress(), StringPool.SEMICOLON,
+				user2.getEmailAddress()));
+
+		List<String> expectedToEmailAddresses2 = ListUtil.sort(
+			Arrays.asList(
+				user2.getEmailAddress(),
+				GetterUtil.getString(
+					childObjectEntryValues.get("emailTextObjectField"))));
+
+		_testSendNotificationWithToProperty(
+			expectedToEmailAddresses2,
+			"[%CURRENT_USER_EMAIL_ADDRESS%]," +
+				getTermName("emailTextObjectField"));
+
+		_testSendNotificationWithToProperty(
+			expectedToEmailAddresses2,
+			"[%CURRENT_USER_EMAIL_ADDRESS%], " +
+				getTermName("emailTextObjectField"));
+
+		_testSendNotificationWithToProperty(
+			expectedToEmailAddresses2,
+			"[%CURRENT_USER_EMAIL_ADDRESS%];" +
+				getTermName("emailTextObjectField"));
```

Comment:
I believe we can make this part more short, and inline some variables attribution:

```java
		for (String separator :
				Arrays.asList(
					StringPool.COMMA, StringPool.COMMA_AND_SPACE,
					StringPool.SEMICOLON)) {

			_testSendNotification(
				ListUtil.sort(
					Arrays.asList(
						user1.getEmailAddress(), user2.getEmailAddress())),
				StringBundler.concat(
					user1.getEmailAddress(), separator,
					user2.getEmailAddress()));

			_testSendNotification(
				ListUtil.sort(
					Arrays.asList(
						user2.getEmailAddress(),
						GetterUtil.getString(
							childObjectEntryValues.get(
								"emailTextObjectField")))),
				StringBundler.concat(
					"[%CURRENT_USER_EMAIL_ADDRESS%]", separator,
					getTermName("emailTextObjectField")));
		}
```